jupyterkernelgen/jupyterkernelgen.py

```diff
@@ -0,0 +1,181 @@
+#!/usr/bin/env python3
+import os
+import glob
+import shutil
+import sys
+import readline
+import subprocess
+
+# allow tab completion in input()
+readline.set_completer_delims(' \t\n=')
+readline.parse_and_bind("tab: complete")
+
+# define ANSI escape codes for coloured text
+class colors:
+    HEADER = '\033[95m'
+    OK = '\033[92m'
+    WARNING = '\033[93m'
+    FAIL = '\033[91m'
+    ENDC = '\033[0m'
+
+def check_for_conda():
+    print(f"{colors.HEADER}LOOKING FOR CONDA...{colors.ENDC}")
+
+    conda_exe = None
+    try:
+        conda_exe = shutil.which("mamba") # first try getting mamba because its faster
+    except shutil.Error as e:
+        print(f"{colors.FAIL}error occurred getting mamba from path: {e}{colors.ENDC}", file=sys.stderr)
+
+    if conda_exe == None:
+        try:
+            conda_exe = shutil.which("conda") # try finding conda if mamba isn't available
+        except shutil.Error as e:
+            print(f"{colors.FAIL}error occurred getting conda from path: {e}{colors.ENDC}", file=sys.stderr)
+            exit(1)
+
+        if conda_exe == None:
+            print(f"{colors.FAIL}no conda executable on the path. Exiting...{colors.ENDC}", file=sys.stderr)
+            sys.exit(1)
+
+    print(f"- found conda executable: {colors.OK}{conda_exe}{colors.ENDC}\n")
+
+    return conda_exe
+
+def get_conda_env():
+    try:
+        # replace ~ with user's home directory and make it an absolute path
+        conda_env = os.path.abspath(os.path.expanduser(input(f"{colors.HEADER}ENTER PATH TO CONDA ENVIRONMENT{colors.ENDC}: ")))
```

Comment:
Assumption that this function will not do anything for the directory is NOT in their home directory ? If so, please add that comment in.

Reply:
correct. If you don't refer to the home directory, it will just make an absolute path without any other changes.